opensir/models/post_regression.py

```diff
@@ -168,19 +259,41 @@ def ci_block_cv(self, lags=1, min_sample=3):
             p_list.append(self.p)
             # Predict for the i + lags period
             self.solve(
-                self.fit_attr["t_obs"][i - 1 + lags],
-                numpoints=int(self.fit_attr["t_obs"][i - 1 + lags]) + 1,
+                self.fit_attr["t_obs"][-1],
+                numpoints=int(self.fit_attr["t_obs"][-1]) + 1,
             )
             pred = self.fetch()[:, self.fit_attr["fit_input"]]
             # Calculate mean squared errors
-            mse = np.sqrt((pred[-1] - self.fit_attr["n_obs"][i - 1 + lags]) ** 2)
-            mse_list.append(mse)
+            if i < len(self.fit_attr["t_obs"]):
+                idx = self.fit_attr["t_obs"][i - 1 + lags :].astype(int)
+                mse = np.sqrt((pred[idx] - self.fit_attr["n_obs"][i - 1 + lags :]) ** 2)
+                mse_fc.append(mse)
 
         p_list = np.array(p_list)
-        mse_list = np.array(mse_list)
-        mse_avg = np.mean(mse_list)
-
-        self.p = p0
-        self.w = w0
-
-        return mse_avg, mse_list, p_list
+        mse_fc = np.array(mse_fc)
+        mse_avg, n_avg, mse_seq = rolling_avg(mse_fc)
+
+        # Generate len(t_obs) - min_sample days predictions
+        self.solve(
+            self.fit_attr["t_obs"][-1] + len(mse_fc),
+            numpoints=int(self.fit_attr["t_obs"][-1]) + len(mse_fc) + 1,
+        )
+
+        # The first element of the prediction is out of the sample
+        t_start = int(self.fit_attr["t_obs"][-1]) + 1
+
+        # pred = self.fetch()[t_start:, self.fit_attr["fit_input"]]
```

Comment:
Why is this code commented out?

Reply:
Because I called the RHS directly on the prediction results.